bread/layout/components/form.py

```diff
@@ -115,89 +120,187 @@ def render(self, context):
             ),
-            hg.DIV(
-                Form(formset.empty_form, *self, standalone=False),
-                id=f"empty_{ formset.prefix }_form",
-                _class="template-form",
-                style="display:none;",
+            # Empty form as template for new entries. The script tag works very well
+            # for this since we need a single, raw, unescaped HTML string
+            hg.SCRIPT(
+                Form(
+                    hg.C(f"{self.formname}.{self.fieldname}.formset.empty_form"),
+                    hg.WithContext(
+                        self.content,
+                        formset_form=hg.C(
+                            f"{self.formname}.{self.fieldname}.formset.empty_form"
+                        ),
+                    ),
+                    standalone=False,
+                ),
+                id=hg.BaseElement(
+                    "empty_",
+                    hg.C(f"{self.formname}.{self.fieldname}.formset.prefix"),
+                    "_form",
+                ),
+                type="text/plain",
             ),
             hg.SCRIPT(
                 mark_safe(
-                    f"""document.addEventListener("DOMContentLoaded", e => init_formset("{ formset.prefix }"));"""
-                )
+                    "document.addEventListener('DOMContentLoaded', e => init_formset('"
+                ),
+                hg.C(f"{self.formname}.{self.fieldname}.formset.prefix"),
+                mark_safe("'));"),
             ),
         )
-        yield from skeleton.render(context)
 
-    def __repr__(self):
-        return f"Formset({self.fieldname}, {self.formsetfactory_kwargs})"
-
-
-class FormsetAddButton(FormChild, Button):
-    def __init__(self, fieldname, label=_("Add"), **kwargs):
+    def add_button(self, container_css_selector, label=_("Add"), **kwargs):
+        prefix = hg.C(f"{self.formname}.{self.fieldname}.formset.prefix")
         defaults = {
             "icon": "add",
             "notext": True,
             "buttontype": "tertiary",
+            "id": hg.BaseElement(
+                "add_", prefix, "_button"
+            ),  # required for javascript to work correctly
+            "onclick": hg.BaseElement(
+                "formset_add('",
+                prefix,
+                "', '",
+                container_css_selector,
+                "');",
+            ),
         }
-        defaults.update(kwargs)
-        self.fieldname = fieldname
-        super().__init__(label, **defaults)
+        return Button(label, **{**defaults, **kwargs})
 
-    def render(self, context):
-        formset = self.form[self.fieldname].formset
-        self.attributes["id"] = f"add_{formset.prefix}_button"
-        self.attributes[
-            "onclick"
-        ] = f"formset_add('{ formset.prefix }', '#formset_{ formset.prefix }_container');"
-        return super().render(context)
+    @staticmethod
+    def as_plain(*args, add_label=_("Add"), **kwargs):
```

Comment:
Maybe a name like `as_textinput_group` would be more descriptive.

Reply:
Why textinput?